src/components/Icon/Icon.tsx

```diff
@@ -1,8 +1,6 @@
 import React from 'react';
 
 import {classNames, variationName} from '../../utilities/css';
-import {isNewDesignLanguageColor} from '../../utilities/color-new-design-language';
```

Comment:
Is this function used anywhere else? If not we should delete its definition too.

Reply:
I think we may still need this and I'm thinking it should be renamed to something like `isValidIconColor` Then we can still use it in the component below. That way if people pass in something like `orange` we can issue a console warning

Reply:
When have props that are strings but only some values are valid we usually just use typescript to assert people are using a valid string. E.g. [Button Size](https://github.com/Shopify/polaris-react/blob/3d2da225b7c484e446dd1522cab0db0dfbfdbb87/src/components/Button/Button.tsx#L31) can be 'small', 'medium' or 'large' and we type it as such.

As far as I remember we've never double checked at runtime too.

I'm proposing that the [`color` prop](https://github.com/Shopify/polaris-react/blob/master/src/types.ts#L57) should have the type `NewDesignLanguageColor` instead of `Color | NewDesignLanguageColor` and that way if people pass in 'orange' then typescript will shout at them, and thus we don't need a runtime check.